physionet-django/project/utility.py

```diff
@@ -249,3 +256,55 @@ def get_form_errors(form):
     for field in form.errors:
         all_errors += form.errors[field]
     return all_errors
+
+def grant_aws_open_data_access(user, project):
+    """
+    Function to grant a AWS ID access to the bukets in the Open Data
+    AWS platform.
+    """
+    url = settings.AWS_CLOUD_FORMATION
+    # The paylod has to be a string in an array
+    payload = {'accountid': ["{}".format(user.cloud_information.aws_id)]}
+    # Custom headers set as a key for a lambda function in AWS to grant access
+    headers = {settings.AWS_HEADER_KEY: settings.AWS_HEADER_VALUE,
+        settings.AWS_HEADER_KEY2: settings.AWS_HEADER_VALUE2}
+    # Do a request to AWS and try to add the user ID to the bucket
+    response = requests.post(url, data=json.dumps(payload), headers=headers)
+    message = response.json()['message'].split(',')[0]
+    # The message can differ if the ID is already there, or non-existent
+    LOGGER.info("AWS message '{0}' for project {1}".format(
+        response.json()['message'], project))
+    return message
+
+def grant_gcp_group_access(user, project, data_access, request):
+    """
+    Funtion to add a specific email address to a organizational google group
+    Returns two things:
+        The first argunet is if acces was awarded.
+        The second argunet is if the access was awarded in a previous time.
+    """
+    email = user.cloud_information.gcp_email.email
+    service = create_directory_service(settings.GCP_DELEGATION_EMAIL)
+    members = service.members().list(groupKey=data_access.location).execute()
+    access = "Access to the GCP BigQuery"
+    if data_access == 3:
```

Comment:
The logic can be difficult to follow in places, so it would be good if we could find ways to simplify or make clearer.